### ISSUE TITLE
anbot

### ISSUE THREAD

**Issue by nesj87** [nesj87/Ai-bot#1]
import streamlit as st
import pandas as pd
import numpy as np
import yfinance as yf
import plotly.graph_objects as go
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

# Titel der App
st.title("🔮 AI Quantifizierungsbot App")

# Sidebar für Benutzereingaben
st.sidebar.header("Einstellungen")
ticker = st.sidebar.text_input("Ticker-Symbol (z. B. AAPL):", "AAPL")
start_date = st.sidebar.date_input("Startdatum:", pd.to_datetime("2020-01-01"))
end_date = st.sidebar.date_input("Enddatum:", pd.to_datetime("2023-01-01"))

# Daten von Yahoo Finance abrufen
@st.cache_data
def load_data(ticker, start, end):
    data = yf.download(ticker, start=start, end=end)
    return data

data = load_data(ticker, start_date, end_date)

# Technische Indikatoren berechnen
def calculate_indicators(df):
    # Gleitender Durchschnitt (20 Tage)
    df['MA_20'] = df['Close'].rolling(window=20).mean()
    # RSI (14 Tage)
    delta = df['Close'].diff()
    gain = (delta.where(delta > 0, 0)).rolling(window=14).mean()
    loss = (-delta.where(delta < 0, 0)).rolling(window=14).mean()
    rs = gain / loss
    df['RSI'] = 100 - (100 / (1 + rs))
    return df

data = calculate_indicators(data)

# Handelssignale generieren
def generate_signals(df):
    df['Signal'] = 0
    df['Signal'] = np.where(df['Close'] > df['MA_20'], 1, 0)
    df['Positions'] = df['Signal'].diff()
    return df

data = generate_signals(data)

# Machine-Learning-Modell für Vorhersagen
def train_ai_model(df):
    # Merkmale und Zielvariable erstellen
    df['Target'] = np.where(df['Close'].shift(-1) > df['Close'], 1, 0)
    features = df[['MA_20', 'RSI']].dropna()
    target = df['Target'].dropna()
    
    # Daten aufteilen
    X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.2, shuffle=False)
    
    # Modell trainieren
    model = RandomForestClassifier(n_estimators=100)
    model.fit(X_train, y_train)
    
    # Vorhersagen treffen
    predictions = model.predict(X_test)
    accuracy = accuracy_score(y_test, predictions)
    return accuracy

accuracy = train_ai_model(data)

# Ergebnisse anzeigen
st.subheader("📊 Finanzdaten")
st.write(data.tail())

# Plot mit Plotly
fig = go.Figure()
fig.add_trace(go.Scatter(x=data.index, y=data['Close'], name='Schlusskurs'))
fig.add_trace(go.Scatter(x=data.index, y=data['MA_20'], name='20-Tage-MA'))
fig.update_layout(title="Preis und gleitender Durchschnitt")
st.plotly_chart(fig)

# Signale anzeigen
st.subheader("🚦 Handelssignale")
st.write("Letzte Signale:")
st.write(data[['Close', 'MA_20', 'Signal']].tail())

# Risikomanagement
st.subheader("📉 Risikomanagement")
volatility = data['Close'].pct_change().std() * np.sqrt(252)
st.metric("Jährliche Volatilität", f"{volatility:.2%}")

# KI-Modell-Performance
st.subheader("🤖 AI-Modell")
st.write(f"Genauigkeit der Vorhersagen: {accuracy:.2%}")

# LSTM-Modell für Zeitreihenvorhersage (Beispiel)
st.subheader("🔮 LSTM-Vorhersagemodell (nur Demo)")
st.warning("Hinweis: LSTM-Modell erfordert GPU und komplexen Code. Dies ist eine vereinfachte Demo.")

# Footer
st.markdown("---")
st.markdown("Erstellt mit ❤️ von AI Bot 🤖 | [GitHub-Repo](https://github.com/)")
